src/ext/Util/ca/scaexec.cpp

```diff
@@ -520,55 +519,83 @@ static HRESULT ModifyUserLocalBatchRight(
     return hr;
 }
 
-static void SetUserPasswordAndAttributes(
-    __in USER_INFO_1* puserInfo,
-    __in LPWSTR wzPassword,
-    __in int iAttributes
-    )
+static void ApplyAttributes(int iAttributes, DWORD* pFlags)
 {
-    Assert(puserInfo);
-
-    // Set the User's password
-    puserInfo->usri1_password = wzPassword;
-
-    // Apply the Attributes
     if (SCAU_DONT_EXPIRE_PASSWRD & iAttributes)
     {
-        puserInfo->usri1_flags |= UF_DONT_EXPIRE_PASSWD;
+        *pFlags |= UF_DONT_EXPIRE_PASSWD;
     }
     else
     {
-        puserInfo->usri1_flags &= ~UF_DONT_EXPIRE_PASSWD;
+        *pFlags &= ~UF_DONT_EXPIRE_PASSWD;
     }
 
     if (SCAU_PASSWD_CANT_CHANGE & iAttributes)
     {
-        puserInfo->usri1_flags |= UF_PASSWD_CANT_CHANGE;
+        *pFlags |= UF_PASSWD_CANT_CHANGE;
     }
     else
     {
-        puserInfo->usri1_flags &= ~UF_PASSWD_CANT_CHANGE;
+        *pFlags &= ~UF_PASSWD_CANT_CHANGE;
     }
 
     if (SCAU_DISABLE_ACCOUNT & iAttributes)
     {
-        puserInfo->usri1_flags |= UF_ACCOUNTDISABLE;
+        *pFlags |= UF_ACCOUNTDISABLE;
     }
     else
     {
-        puserInfo->usri1_flags &= ~UF_ACCOUNTDISABLE;
+        *pFlags &= ~UF_ACCOUNTDISABLE;
     }
 
     if (SCAU_PASSWD_CHANGE_REQD_ON_LOGIN & iAttributes) // TODO: for some reason this doesn't work
     {
-        puserInfo->usri1_flags |= UF_PASSWORD_EXPIRED;
+        *pFlags |= UF_PASSWORD_EXPIRED;
     }
     else
     {
-        puserInfo->usri1_flags &= ~UF_PASSWORD_EXPIRED;
+        *pFlags &= ~UF_PASSWORD_EXPIRED;
     }
 }
 
+static void ApplyComment(int iAttributes, LPWSTR pwzComment, LPWSTR* ppComment)
+{
+    if (SCAU_REMOVE_COMMENT & iAttributes)
+    {
+        *ppComment = L"";
+    }
+    else if (pwzComment && *pwzComment)
+    {
+        *ppComment = pwzComment;
+    }
+}
+
+static NET_API_STATUS SetUserPassword(__in LPWSTR pwzServerName, __in LPWSTR pwzName, __in LPWSTR pwzPassword)
+{
+    DWORD dw;
```

Comment:
This isn't used, so you can pass `NULL` to `NetUserSetInfo()` and avoid creating it.

Reply:
Done.